platform/efm32/inc/configurations.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 Silicon Laboratories, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __CONFIGURATIONS_H__
+#define __CONFIGURATIONS_H__
+
+/*******************************************************************************
+ * Universal configurations
+ ******************************************************************************/
+
+/* Memory boundaries */
+#define FLASH_ORIGIN 0x00000000
+
+/* Memory boundaries */
+#define SRAM_ORIGIN  0x20000000
+#define SRAM_OFFSET  0x0
+
+/* Minimum memory requirements */
+#define FLASH_LENGTH_MIN 0x10000  /* 64k */
+#define SRAM_LENGTH_MIN  0x8000   /* 32k */
+
+/*******************************************************************************
+ * Hardware-specific configurations
+ *
+ * Configurations are named after the parameter values, in this order:
+ *   - CORE
+ *   - PLATFORM
+ ******************************************************************************/
+#if defined(CONFIGURATION_EFM32_M3_P1)
+    /* ARM core selection */
+#   define CORE_CORTEX_M3
```

Comment:
Could you move `NVIC_VECTORS` and `FLASH_OFFSET` under "Universal configurations"? Even if there might be future configurations that require to have differentiated values for those symbols, I'd like to have current configurations only rely on the differentiating one.

As a general approach, I think `NVIC_VECTORS` will always be one single value across a family (defined as the max of all values across the family). An exception might be families with such a large variance that might benefit in having two configurations (for wasting less memory). For example:

> In the family, the following number of NVIC vectors are available: 17, 19, 21, 60, 76, 84. I'd have two configurations with `NVIC_VECTORS` set to 21 (for {17, 19, 21}) and 84 (for {60, 76, 84}), respectively (local max on a cluster).